### PR TITLE
Update include path support

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge accomplishes 3 main goals:
- Remove support for `application_directory` within workspace configuration files
- Enhance support for `include` files / paths in workspace configuration files
- Update documentation and tests to ensure includes work as expected.

### CODE REVIEW

File: lib/ramble/ramble/workspace/workspace.py [hunk 1]
```diff
@@ -1639,6 +1619,24 @@ def all_auxiliary_software_files(self):
         """Iterator over each file in $workspace/configs/auxiliary_software_files"""
         yield from self._auxiliary_software_files.items()
 
+    def workspace_paths(self):
+        """Dictionary of path replacements for workspace"""
+        if not hasattr(self, "_workspace_path_replacements"):
+            self._workspace_path_replacements = {
```
Comment (linsword13) [inline lib/ramble/ramble/workspace/workspace.py:1625]: Do we want to document these keywords?

**Review by linsword13 — COMMENTED** [review]
Great stuff! Want to confirm as you mentioned you updated tests, but I don't see changes there?

File: lib/ramble/ramble/workspace/workspace.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/workspace/workspace.py:1625]: Thanks, I did mean to do that but forgot to.

**Review by douglasjacobsen — COMMENTED** [review]
Ah! Good catch. I didn't push the last commit which added the test.

**Review by linsword13 — APPROVED** [review]

**Pull request merged**